security/ssh_certificate.ksy

```diff
@@ -0,0 +1,118 @@
+meta:
+  id: ssh_certificate
+  title: OpenSSH Certificate
+  license: CC0-1.0
+  endian: be
+  imports:
+    - /security/ssh_public_key
+doc: |
+  OpenSSH Certificates are simple certs used by OpenSSH.
+doc-ref: 'https://cvsweb.openbsd.org/src/usr.bin/ssh/PROTOCOL.certkeys?annotate=HEAD'
+seq:
+  - id: cert_name
+    type: ssh_public_key::cstring
+  - id: body
+    type:
+      switch-on: cert_name.value
+      cases:
+        '"ssh-rsa-cert-v01@openssh.com"': cert_rsa
+        '"ssh-dss-cert-v01@openssh.com"': cert_dss
+        '"ecdsa-sha2-nistp256-cert-v01@openssh.com"': cert_ecdsa
+        '"ecdsa-sha2-nistp384-cert-v01@openssh.com"': cert_ecdsa
+        '"ecdsa-sha2-nistp521-cert-v01@openssh.com"': cert_ecdsa
+        '"ssh-ed25519-cert-v01@openssh.com"': cert_ed25519
+types:
+  cert_rsa:
+    doc-ref: 'https://tools.ietf.org/html/rfc4253#section-6.6'
+    seq:
+      - id: nonce
+        type: ssh_public_key::cstring
```

Comment:
I don't think that `nonce` is an ASCII string (see [`security/ssh_public_key.ksy:84`](https://github.com/kaitai-io/kaitai_struct_formats/blob/7f3ecee038ae4aa14da7d9a80d8abfd4080c9b59/security/ssh_public_key.ksy#L84)).

When I generate a test file with these commands:

```shell
ssh-keygen -t rsa -f ./id_rsa -N ""
ssh-keygen -s ./id_rsa -I "AcmeCorporation" ./id_rsa.pub
cat ./id_rsa-cert.pub | cut -f2 -d ' ' | openssl base64 -d -A > id_rsa-cert_base64-decoded.bin
```

it seems that this `nonce` is just an array of random bytes, so it should be treated that way. Interpreting it as a string with encoding is useless. Note that the `ASCII` encoding used isn't even guaranteed to properly handle byte values 128-255.

Reply:
Yes, you are correct, I had moved to using the imported type without noticing that.. for reference the RFC text is as follows re: openssh "strings".
```
   string

      Arbitrary length binary string.  Strings are allowed to contain
      arbitrary binary data, including null characters and 8-bit
      characters.  They are stored as a uint32 containing its length
      (number of bytes that follow) and zero (= empty string) or more
      bytes that are the value of the string.  Terminating null
      characters are not used.

      Strings are also used to store text.  In that case, US-ASCII is
      used for internal names, and ISO-10646 UTF-8 for text that might
      be displayed to the user.  The terminating null character SHOULD
      NOT normally be stored in the string.  For example: the US-ASCII
      string "testing" is represented as 00 00 00 07 t e s t i n g.  The
      UTF-8 mapping does not alter the encoding of US-ASCII characters.
```
I will add a non-ASCII type for use here for the fields that don't actually store strings.